Manito/Manito/Screens/Letter/ViewModels/LetterViewModel.swift

```diff
@@ -0,0 +1,182 @@
+//
+//  LetterViewModel.swift
+//  Manito
+//
+//  Created by SHIN YOON AH on 2023/06/20.
+//
+
+import Combine
+import Foundation
+
+final class LetterViewModel: ViewModelType {
+
+    typealias MessageDetails = (roomId: String, mission: String, missionId: String, manitteeId: String)
+    typealias ReportDetails = (nickname: String, content: String)
+
+    enum MessageType: Int {
+        case sent = 0
+        case received = 1
+    }
+
+    enum RoomState: String {
+        case processing = "PROCESSING"
+        case post = "POST"
+    }
+
+    struct Input {
+        let viewDidLoad: AnyPublisher<Void, Never>
+        let segmentControlValueChanged: PassthroughSubject<Int, Never>
+        let refresh: PassthroughSubject<Void, Never>
+        let sendLetterButtonDidTap: AnyPublisher<Void, Never>
+        let reportButtonDidTap: PassthroughSubject<String, Never>
+    }
+
+    struct Output {
+        let messages: PassthroughSubject<[Message], NetworkError>
+        let index: CurrentValueSubject<Int, Never>
+        let messageDetails: AnyPublisher<MessageDetails, Never>
+        let reportDetails: AnyPublisher<ReportDetails, Never>
+        let roomState: AnyPublisher<RoomState, Never>
+    }
+
+    // MARK: - property
+
+    private let messageSubject: PassthroughSubject<[Message], NetworkError> = PassthroughSubject()
+    private lazy var indexSubject: CurrentValueSubject<Int, Never> = CurrentValueSubject(self.messageType.rawValue)
+
+    private var cancelBag: Set<AnyCancellable> = Set()
+
+    private let service: LetterServicable
+    private var messageDetails: MessageDetails
+    private let roomState: RoomState
+    private let messageType: MessageType
+
+    // MARK: - init
+
+    init(service: LetterServicable,
+         roomId: String,
+         mission: String,
+         missionId: String,
+         roomState: String,
+         messageType: MessageType) {
+        self.service = service
+        self.messageDetails = MessageDetails(roomId, mission, missionId, "")
+        self.roomState = RoomState(rawValue: roomState)!
+        self.messageType = messageType
+    }
+
+    // MARK: - Public - func
+
+    func transform(from input: Input) -> Output {
+        let viewDidLoadType = input.viewDidLoad
+            .map { [weak self] in self?.messageType }
+            .map { $0! }
```

Comment:
p3;

여기서 compactMap을 사용한다면 아래의 map에서 강제 언래핑을 사용하지 않을 수 있을 것 같아요!

```suggestion
            .compactMap { [weak self] in self?.messageType}
            .map { $0 }
```

Reply:
[weak self] 때문에 죄다 언래핑 해야된다길래 짱나서 냅다 느낌표만 달았네요. 죄송합니다^^

[ef3c0b0](https://github.com/DeveloperAcademy-POSTECH/MC2-Team5-Firefighter/pull/466/commits/ef3c0b0eff054cd98be5d4b48b115b4643d2d10b)